contracts/bprotocol/scoring/BTokenScore.sol

```diff
@@ -33,48 +69,125 @@ contract BTokenScore is ScoringMachine {
         return keccak256(abi.encodePacked(parent, "coll", cToken));
     }
 
-    function slashedAsset(address cToken) public pure returns (bytes32) {
+    function slashedDebtAsset(address cToken) public pure returns (bytes32) {
         return keccak256(abi.encodePacked(parent, "slashed-debt", cToken));
     }
 
+    function slashedCollAsset(address cToken) public pure returns (bytes32) {
+        return keccak256(abi.encodePacked(parent, "slashed-coll", cToken));
+    }
+
     // Update Score
     // =============
-    function updateDebtScore(address _user, address cToken, int256 amount) external onlyAvatar {
-        updateScore(user(_user), debtAsset(cToken), amount, now);
+    function updateDebtScore(address _avatar, address cToken, int256 amount) external onlyAvatar {
+        updateScore(user(_avatar), debtAsset(cToken), amount, now);
+    }
+
+    function updateCollScore(address _avatar, address cToken, int256 amount) external onlyAvatar {
+        updateScore(user(_avatar), collAsset(cToken), amount, now);
+    }
+
+    function mul_(uint score, uint multiplier, uint224 index) internal pure returns (uint) {
+        return mul_(mul_(score, multiplier), index);
     }
 
-    function updateCollScore(address _user, address cToken, int256 amount) external onlyAvatar {
-        updateScore(user(_user), collAsset(cToken), amount, now);
+    function updateIndex(address cToken) public {
+        require(expiry < now, "expired");
+        uint224 supplyIndex;
+        uint224 borrowIndex;
+        (supplyIndex,) = comptroller.compSupplyState(cToken);
+        (borrowIndex,) = comptroller.compBorrowState(cToken);
+        lastIndex[cToken].supplyIndex = supplyIndex;
+        lastIndex[cToken].borrowIndex = borrowIndex;
     }
 
-    function slashedScore(address _user, address cToken, int256 amount) external {
-        // TODO implement
-        // updateScore(user(_user), slashedAsset(cToken), amount, now);
+    function _getDeltaSupplyIndex(address cToken) internal returns (uint224 deltaSupplyIndex) {
+        uint224 currSupplyIndex;
+        (currSupplyIndex,) = comptroller.compSupplyState(cToken);
+        uint deltaSupplyIndexForCToken = sub_(uint(currSupplyIndex), uint(lastIndex[cToken].supplyIndex));
+
+        // NOTICE: supplyIndex takes cToken.totalSupply() which is in cToken quantity
+        // We need the index normalized to underlying token quantity
+        uint exchangeRate = ICToken(cToken).exchangeRateCurrent();
+        deltaSupplyIndex = safe224(div_(deltaSupplyIndexForCToken, exchangeRate), "index-exceeds-224-bits");
```

Comment:
are the resolutions here ok?i think exchange rate is 1e18, shouldn't we also multiply by something?